sources/RuleRunner.cpp

```diff
@@ -1329,6 +1330,96 @@ ActionHasTarget(int8 type)
 }
 
 
+static bool
+getDirectoryPath(BString& str, const entry_ref& ref)
+{
+	BEntry entry(&ref);
+	if (entry.InitCheck() != B_OK)
+		return false;
+
+	if (!entry.IsDirectory() && entry.GetParent(&entry) != B_OK)
+		return false;
+
+	BPath path(&entry);
+	if (path.InitCheck() != B_OK)
+		return false;
+
+	str = path.Path();
+	return true;
+}
+
+
+bool
+SetTextForType(BString& text, int8 type, const entry_ref& ref, bool isTest)
+{
+	if (isTest)
+		switch (type) {
+			case TEST_TYPE:
+			{
+				BNode node(&ref);
+				if (node.InitCheck() != B_OK)
+					return false;
+
+				BNodeInfo nodeInfo(&node);
+				if (nodeInfo.InitCheck() != B_OK)
+					return false;
+
+				char mimeType[B_MIME_TYPE_LENGTH];
+				if (nodeInfo.GetType(mimeType) != B_OK)
+					return false;
+
+				text = mimeType;
+				return true;
+			}
+			case TEST_NAME:
+				text = ref.name;
+				return true;
+			case TEST_SIZE:
+			{
+				BEntry entry(&ref);
+				if (entry.InitCheck() != B_OK || !entry.IsFile())
+					return false;
+
+				off_t size;
+				if (entry.GetSize(&size) != B_OK)
+					return false;
+
+				text = "";
+				text << size;
+				return true;
+			}
+			case TEST_LOCATION:
+				return getDirectoryPath(text, ref);
+			default:
+				return false;
+		}
+	else
+		switch (type) {
+			case ACTION_MOVE:
+			case ACTION_COPY:
+				return getDirectoryPath(text, ref);
+			case ACTION_RENAME:
+				text = ref.name;
+				return true;
+			case ACTION_ARCHIVE:
+			{
+				BEntry entry(&ref);
+				if (entry.InitCheck() != B_OK || !entry.IsFile())
+					return false;
+
+				BPath path(&ref);
+				if (path.InitCheck() != B_OK)
+					return false;
+
+				text = path.Path();
+				return true;
```

Comment:
Is Archive.zip the standard Haiku archive file name? If we localize it, do we have to set standard archive names for other locales or let the translators decide? Should we localize only the base name Archive or both the base name and the suffix zip? Because so far no file names or text control values are localized, I think we probably just want to append "/Archive.zip" to the drag&dropped folder path. (And not worrying about whether or not the file already exists because it should be handled by the invoked zip command.) What do you think?